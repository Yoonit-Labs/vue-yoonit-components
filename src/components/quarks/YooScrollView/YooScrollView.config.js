module.exports = {
  orientation: {
    options: [
      'vertical',
      'horizontal'
    ]
  }
}
