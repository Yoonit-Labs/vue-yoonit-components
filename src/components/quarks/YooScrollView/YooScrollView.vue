<template lang="pug">
  .yoo-scroll-view(
    :class="takeOrientation"
  )
    slot
</template>

<script>

export default {
  name: 'YooScrollView',
  props: {
    orientation: {
      type: String,
      required: false,
      default: 'vertical',
      validator: value => [
        'vertical',
        'horizontal'
      ]
        .includes(value)
    }
  },
  components: {},
  computed: {
    takeOrientation () {
      return `yoo-scroll-view--${this.orientation}`
    }
  },
  methods: {},
  filters: {},
  watch: {}
}
</script>
