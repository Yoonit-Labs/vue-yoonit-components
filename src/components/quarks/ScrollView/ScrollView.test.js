import { shallowMount } from '@vue/test-utils'
import YooScrollView from '@/components/quarks/ScrollView/ScrollView.vue'
import PropsConfig from '@/components/quarks/ScrollView/ScrollView.config'

const classBlock = 'yoo-scroll-view'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooScrollView, {
    slots: { default: SlotText }
  })
}

describe('YooIndicator Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.find('.yoo-scroll-view').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('orientation', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.orientation.options.includes(YooScrollView.props.orientation.default)).toBe(true)
      })
      PropsConfig.orientation.options.forEach(orientation => {
        it(`Includes orientation class: .${classBlock}--${orientation}`, async () => {
          await wrapper.setProps({ orientation })
          expect(wrapper.find(`.${classBlock}--${orientation}`).exists()).toBe(true)
        })
      })
    })
  }) // describe Props
})
