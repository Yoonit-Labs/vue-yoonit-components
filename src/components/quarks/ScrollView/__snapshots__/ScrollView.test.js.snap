// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooIndicator Component Matches Snapshot 1`] = `<div class="yoo-scroll-view yoo-scroll-view--vertical">Default Slot Text</div>`;
